client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -409,7 +412,27 @@ public void readLog(PluginCall call) {
         }
         call.resolve(response);
     }
-   
+
+    @PluginMethod()
+    public void acquireWakeLock(PluginCall call){
+        try {
+            PowerManager powerManager = (PowerManager) getContext().getSystemService(POWER_SERVICE);
+            wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "omSupply::WakelockTag");
+            wakeLock.acquire();
+        } catch (Exception e) {
+            Log.e(OM_SUPPLY, "Error acquiring wakelock: " + e.getMessage() );
+        }
+    }
+
+    @PluginMethod()
+    public void releaseWakeLock(PluginCall call){
+        try {
+            wakeLock.release();
```

Comment:
do we need to check if it's null, and do we need to set it to null after released ? Or is it ok for this to be called in an error.

I am just theorising, but to minimise the chance of concurrent wakeLocks, we could only acquire lock when it's been released (not null), and only release if it's not null and set to null on release